src/api/server.rkt

```diff
@@ -85,62 +85,61 @@
 
 ; Returns #f is now job exsist for the given job-id
 (define (get-results-for job-id)
-  (define-values (a b) (place-channel))
-  (place-channel-put manager (list 'result job-id b))
   (log "Getting result for job: ~a.\n" job-id)
-  (place-channel-get a))
+  (manager-ask 'result job-id))
 
 (define (get-timeline-for job-id)
-  (define-values (a b) (place-channel))
-  (place-channel-put manager (list 'timeline job-id b))
   (log "Getting timeline for job: ~a.\n" job-id)
-  (place-channel-get a))
+  (manager-ask 'timeline job-id))
 
 ; Returns #f if there is no job returns the job-id if there is a completed job.
 (define (server-check-on job-id)
-  (define-values (a b) (place-channel))
-  (place-channel-put manager (list 'check job-id b))
   (log "Checking on: ~a.\n" job-id)
-  (place-channel-get a))
+  (manager-ask 'check job-id))
 
 (define (get-improve-table-data)
-  (define-values (a b) (place-channel))
-  (place-channel-put manager (list 'improve b))
   (log "Getting improve results.\n")
-  (place-channel-get a))
+  (manager-ask 'improve))
 
 (define (job-count)
-  (define-values (a b) (place-channel))
-  (place-channel-put manager (list 'count b))
-  (define job-list (place-channel-get a))
-  (log "Currently ~a jobs in progress, ~a jobs in queue." (first job-list) (second job-list))
+  (define job-list (manager-ask 'count))
+  (log "Currently ~a jobs in progress, ~a jobs in queue.\n" (first job-list) (second job-list))
   (apply + job-list))
 
 ;; Starts a job for a given command object|
 (define (start-job command)
   (define job-id (compute-job-id command))
-  (place-channel-put manager (list 'start manager command job-id))
+  (manager-ask 'start manager command job-id)
   (log "Job ~a, Qed up for program: ~a\n" job-id (test-name (herbie-command-test command)))
   job-id)
 
 (define (wait-for-job job-id)
-  (define-values (a b) (place-channel))
-  (place-channel-put manager (list 'wait manager job-id b))
-  (define finished-result (place-channel-get a))
+  (define finished-result (manager-ask 'wait manager job-id))
   (log "Done waiting for: ~a\n" job-id)
   finished-result)
 
-; TODO refactor using this helper.
 (define (manager-ask msg . args)
-  (define-values (a b) (place-channel))
-  (place-channel-put manager (cons msg b args))
   (log "Asking manager: ~a, ~a.\n" msg args)
+  (match msg
+    ['wait (manager-ask-with-callback msg args)]
+    ['result (manager-ask-with-callback msg args)]
+    ['timeline (manager-ask-with-callback msg args)]
+    ['check (manager-ask-with-callback msg args)]
+    ['count (manager-ask-with-callback msg args)]
+    ['improve (manager-ask-with-callback msg args)]
+    [_ (place-channel-put manager (flatten (list msg args)))]))
```

Comment:
AFAICT the only difference between the cases is whether or not you send a reply channel. Can we just ignore that reply channel in the else case, whatever it's for?

Reply:
By reply channel I mean the `b` part of `(place-channel)` in `manager-ask-with-callback`

Reply:
Oh, I see—we can't because we don't always wait on the callback. Fine.

Reply:
You could imagine having two functions, `manager-ask` and `manager-tell`, for the callback and no-callback case, instead of having this big `match` statement here.